### PR TITLE
Issue #11584: fix WriteTag default configuration reporting

### LINKED ISSUE #11584: WriteTag reports violation with confusing message when there is no javadoc
https://checkstyle.org/config_javadoc.html#WriteTag
```
$ cat Test.java
public class Test {}

$ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WriteTag"/>
    </module>
</module>

$ java -jar checkstyle-10.2-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\workdir\Test.java:1: Type Javadoc comment is missing null tag. [WriteTag]
Audit done.
Checkstyle ends with 1 errors.
```

Expected: no violation, since there is no javadoc.
Also note confusing violation message ("null" tag).

For javadoc presence there are other checks. And docs mention that with default config it does nothing, but it reports weird message instead.

P.S. Checkstyle has `WriteTag` with default config - 
https://github.com/checkstyle/checkstyle/blob/4ad253653d59d77002ae2f61349624dcb18dcc38/config/checkstyle_checks.xml#L653

--------

**Breaking compatibility migration notes:**
after fix and release (10.21.0) Check will not report violations on missed Javadocs.
To archive behavior as before, users should start using [InvalidJavadocPosition](https://checkstyle.org/checks/javadoc/invalidjavadocposition.html), [MissingJavadocMethod](https://checkstyle.org/checks/javadoc/missingjavadocmethod.html) , [MissingJavadocPackage](https://checkstyle.org/checks/javadoc/missingjavadocpackage.html) , [MissingJavadocType](https://checkstyle.org/checks/javadoc/missingjavadoctype.html) to let Checkstyle demand javadoc first and when WriteTag will  demand required tag in existing javadoc

### PR DESCRIPTION
**Pull request by strkkk**
Closes #11584 
Just check it is not null

Diff regression config: https://gist.githubusercontent.com/strkkk/238b6c97492ff6ffdb3be1603deca8fc/raw/5db22461435a94c56a3f420be844cd07baf497f2/config_single_2.xml

### CODE REVIEW

**Comment by romani** [comment]
`[WARNING] Rule violated for class com.puppycrawl.tools.checkstyle.checks.javadoc.WriteTagCheck: branches covered ratio is 0.93, but expected minimum is 1.00`

@strkkk , please add test coverage.

**Review by romani — CHANGES_REQUESTED** [review]
code is ok, CI failures must be updated

**Comment by strkkk** [comment]
Github, generate report

**Review by romani — APPROVED** [review]
Ok to merge.

As I come to laptop I can create new PR from my name

**Comment by romani** [comment]
I just rebased by Github .... lets see if this helps

**Comment by romani** [comment]
and I see CircleCI executions :).
So all maintainers can help strkkk with CI.

**Comment by romani** [comment]
@rnveach , please finish review.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheck.java [hunk 1]
```diff
@@ -284,7 +284,9 @@ public void visitToken(DetailAST ast) {
         final TextBlock cmt =
             contents.getJavadocBefore(lineNo);
         if (cmt == null) {
-            log(lineNo, MSG_MISSING_TAG, tag);
+            if (tag != null) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheck.java:287]: > Also note confusing violation message ("null" tag).

@strkkk @romani So are we not resolving this in this issue or is this already another issue?

**Review by rnveach — COMMENTED** [review]
https://checkstyle.org/config_javadoc.html#WriteTag_Description
It is not really clear from descriptions that we validate things with no javadoc. Can we expand that in this PR in the description and examples?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheck.java:287]: We can do in this PR or separate , up to PR author.

Fyi, we are very close to merge at https://github.com/checkstyle/checkstyle/pull/12945

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheck.java:287]: We can do in this PR or separate , up to PR author.

Fyi, we are very close to merge at https://github.com/checkstyle/checkstyle/pull/12945

Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheck.java]: Sorry, my bad. Actually I fixed the message stuff but forgot about skipping javadoc. 

Now it is fixed properly. Weird message with null wont be reported as well.

**Comment by strkkk** [comment]
@rnveach I added a note to description and extended the first example

File: src/xdocs/config_javadoc.xml [hunk 2]
```diff
@@ -4444,6 +4445,9 @@ public class Test {
 public class Test { // violation as required tag is missed
   /** some doc */
   void foo() {} // OK, as methods are not checked by default
+}
+
+class Test2 { // OK, no javadoc
```
Comment (romani) [inline src/xdocs/config_javadoc.xml:4450]: All java examples should be same. Make it single line and copy in all other examples

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheck.java [hunk 3]
```diff
@@ -283,10 +287,7 @@ public void visitToken(DetailAST ast) {
         final int lineNo = ast.getLineNo();
         final TextBlock cmt =
             contents.getJavadocBefore(lineNo);
-        if (cmt == null) {
-            log(lineNo, MSG_MISSING_TAG, tag);
-        }
-        else {
+        if (cmt != null) {
             checkTag(lineNo, cmt.getText());
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheck.java:291]: > Now it is fixed properly. Weird message with null wont be reported as well.

Explain how weird message won't be reported when the log statement is in checkTag and there is nothing between here and there to skip logging if `tag` is `null`?

If we are skipping all violations when `tag` is `null` then we need to make it clear in documentation that tag must be specified.

**Review by rnveach — CHANGES_REQUESTED** [review]

**Comment by romani** [comment]
@strkkk , please resolve conflict

**Comment by romani** [comment]
@strkkk , do you have time to finish this PR?

**Comment by romani** [comment]
I got connection to author, but he can not finish this PR, so everyone is welcome to finish this update.

**Comment by romani** [comment]
@aayushRedHat , please help us to finish this PR, looks like a little is left to be done, but still needs to be done.

**Comment by romani** [comment]
Closed in favor of https://github.com/checkstyle/checkstyle/pull/14153

**Pull request closed without merging**